docs/resource-providers/index.md

```diff
@@ -50,6 +50,31 @@ class DynamoDbTableResourceProvider implements ResourceProvider<ITable> {
     }
 }
 
+/**
+ * Example implementation of a VPC Provider that creates a NAT Gateway 
+ * which is available in all 3 AZs of the VPC while only being in one
+ */
+class OtherVpcResourceProvider implements ResourceProvider<IVpc> {
```

Comment:
@zjaco13 this is the framework repo, it is different from examples repo. That means that code here is not an example and must configurable, reusable, etc. 

It also must be tested, we have unit tests and e2e tests. 

There are a couple of ways to approach this task:

1. Move the example resource provider (Let's call CustomVpcResourceProvider) to the patterns dir. It can stay as is but we should still use it somewhere in a pattern to validate. 
2. Move this code to documentation as example code.
3. Improve the existing VPC resource provider to become more configurable, such as accept subnet configuration, availability zones. 

@elamaran11 I am a bit concerned with this example as it won't register the provisioned subnets to be reused with vpc-cni. So if this provider is used customers won't be able to use secondary ENIs. Thoughts?

Reply:
Currently this is in the resources providers documentation, is there a better place in the docs to put it?

Reply:
I do think it should be more reusable, but I'm struggling to figure out the best way to go about it.  Would it make sense to define a construct and properties that include these VPC properties?

Reply:
All looks good. I must have assume it was code that you added to the repo not docs.